appserver/web/war-util/src/main/java/org/glassfish/web/loader/WebappClassLoader.java

```diff
@@ -3271,19 +3217,19 @@ private boolean isDelegateFirstResource(String name) {
         if (delegate) {
             return true;
         }
-        if (name.startsWith("javax") || name.startsWith("jakarta")) {
-            return !useMyFaces && !name.startsWith("jakarta/faces");
+        if (name.startsWith("jakarta/faces") || name.startsWith("com/sun/faces")) {
+            return !useMyFaces;
         }
-        if (name.startsWith("com/sun/faces")) {
-            return !useMyFaces && !name.startsWith("com/sun/faces/extensions");
+        if (name.startsWith("jakarta") || name.startsWith("javax")) {
+            return true;
         }
-        if (name.startsWith("org/apache/taglibs/standard")) {
```

Comment:
@dmatej Do you remember the reason why this line was removed?

Reply:
The reason is that I thought that these resources moved to jakarta.* packages, but now I see that was just api, impl is in jakarta.servlet.jsp.jstl-3.0.0.jar and that still uses the `org.apache.taglibs` package. However ...

- in those packages are no resources except classes and resource bundles.
- all TCK and GlassFish tests passed
- tld files are under the path `META-INF/*.tld`, so then this delegation filter still wasn't applied

What do you think, is it alright or should we put these "exceptions" back? Did you notice some issue? If I would put that back, at least I would like to document why is it there.

Reply:
@dmatej 
Maybe there is no problem so far.
However, inconsistencies in the following package list may cause problems in the future I think:

- for classes
https://github.com/eclipse-ee4j/glassfish/blob/f2dd7ea034f467b9c96c808c4ae6c68bd597b542/appserver/web/war-util/src/main/java/org/glassfish/web/loader/WebappClassLoader.java#L145-L156

- for resources
https://github.com/eclipse-ee4j/glassfish/blob/f2dd7ea034f467b9c96c808c4ae6c68bd597b542/appserver/web/war-util/src/main/java/org/glassfish/web/loader/WebappClassLoader.java#L3216-L3230

Reply:
Yeah, I noticed inconsistencies even in the original code, it's quite confusing what is useful, what was useful 15 years ago, and what is even harmful. I would like to have more tests for that as it seems to me it was created iteratively just when some TCK test failed or some customer complained that something is not working as he expected. 

That myFaces is another example. Why was the attribute created? What about PrimeFaces and BootsFaces? And is it still useful in GlassFish7?

Reply:
I think now I have the answer in the next comment, thanks!